### PR TITLE
Query update animals table

### PR DESCRIPTION
**Pull request by anagudelogu**
# In this Milestone I:
## Checked the following tasks:

- [x] Add a column species of type `string` to your `animals` table.
- [x] Insert the required data.
- [x] Inside a transaction `update` the `animals` table by setting the `species` column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.

- [x] Inside a transaction:
   - Update the `animals` table by setting the species column to `digimon` for all animals that have a name ending in mon.
   - Update the `animals` table by setting the species column to pokemon for all animals that don't have species already set.
   - Commit the transaction.

- [x] Inside a transaction delete all records in the animals table, then roll back the transaction.

- [x] Inside a transaction:
   - Delete all animals born after Jan 1st, 2022.
   - Create a savepoint for the transaction.
   - Update all animals' weight to be their weight multiplied by -1.
   - Rollback to the savepoint
   - Update all animals' weights that are negative to be their weight multiplied by -1.
   - Commit transaction

- [x] Write queries to answer the following questions:
   - How many animals are there?

   - How many animals have never tried to escape?

   - What is the average weight of animals?

   - Who escapes the most, neutered or not neutered animals?

   - What is the minimum and maximum weight of each type of animal?

   - What is the average number of escape attempts per animal type of those born between 1990 and 2000?

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -33,4 +33,30 @@ WHERE NOT name = 'Gabumon';
...
+SELECT species,
+  MIN(weight_kg) as MIN_WEIGHT,
+  MAX(weight_kg) as MAX_WEIGHT
+FROM animals
+GROUP BY species;
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT species,
+  AVG(escape_attempts) as Average_escape_attempts
+FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
```
Comment (AnselemOdims) [inline queries.sql:107]: - [x] Please kindly include queries that make `UPDATE` and `DELETE` transactions as specified in the requirements

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @anagudelogu , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: queries.sql (hunk 1 above)
Comment (anagudelogu) [reply queries.sql:107]: Sure, will do! Thank you. @AnselemOdims

File: queries.sql [hunk 2]
```diff
@@ -33,4 +33,76 @@ WHERE NOT name = 'Gabumon';
...
+-- DAY 2
+-- Update the animals table by setting the species column to unspecified.
+-- Verify that change was made. Then roll back the change
+-- Verify that the species columns went back to the state before the transaction.
+BEGIN;
+UPDATE animals
+SET species = 'unspecified';
+TABLE animals;
+ROLLBACK;
+TABLE animals;
```
Comment (AnselemOdims) [inline queries.sql:46]: - [ ] Please kindly ensure to include queries to verify that the change was made before and after rollback. In this case, you can do something like this
`SELECT species from animals; `

File: queries.sql [hunk 3]
```diff
@@ -33,4 +33,76 @@ WHERE NOT name = 'Gabumon';
...
+-- Commit the transaction.
+BEGIN;
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+TABLE animals;
+COMMIT;
```
Comment (AnselemOdims) [inline queries.sql:58]: - [ ] As pointed out above, kindly include verification queries

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @anagudelogu , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: queries.sql (hunk 2 above)
Comment (anagudelogu) [reply queries.sql:46]: @AnselemOdims TABLE animals;
prints the entire table with the new updates, I used that as my verification query.
isn't it valid?

**Review by rukundoeric — APPROVED** [review]
Hello @anagudelogu 👋,

_This is an additional review as a result of the second opinion request._

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_

As this was the only reason that your project was not approved it will be approved as soon as possible.
Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approve right away :)

Keep rocking 🚀!

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @anagudelogu, 

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

-----
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**